editor/modes/visual-editor/block-list.js

```diff
@@ -133,14 +133,15 @@ class VisualEditorBlockList extends Component {
 		const boundaries = this.nodes[ uid ].getBoundingClientRect();
 
 		// Create a uid to Y coördinate map.
-		const uidToCoordMap = mapValues( this.nodes, ( node ) => {
-			return node.getBoundingClientRect().top - boundaries.top;
-		} );
+		const uidToCoordMap = mapValues( this.nodes, ( node ) =>
+			node.getBoundingClientRect().top - boundaries.top );
 
 		// Cache a Y coördinate to uid map for use in `onPointerMove`.
 		this.coordMap = invert( uidToCoordMap );
 		// Cache an array of the Y coördinates for use in `onPointerMove`.
-		this.coordMapKeys = Object.values( uidToCoordMap );
+		// Sort the coördinates, as `this.nodes` will not necessarily reflect
+		// the current block sequence.
+		this.coordMapKeys = sortBy( Object.values( uidToCoordMap ) );
```

Comment:
Hm, aren't they? This is recalculated every time at selection start, no?

Reply:
`coordMapKeys` and the other temporary variables in this method are recalculated, but they all derive from `this.nodes`, which is only changed in `setBlockRef`. If you inspect `setBlockRef` calls, you'll see it doesn't fire often (e.g. new block insertion). Namely, `this.nodes` stays the same when moving blocks. And that makes sense: as an [unordered] set of nodes, it still reflects the current post, we just can't expect the keys in the map to be in a particular order.